hedera-node/src/main/java/com/hedera/services/ledger/interceptors/AccountsCommitInterceptor.java

```diff
@@ -54,13 +79,32 @@ public AccountsCommitInterceptor(final SideEffectsTracker sideEffectsTracker) {
 	 */
 	@Override
 	public void preview(final EntityChangeSet<AccountID, MerkleAccount, AccountProperty> pendingChanges) {
+		// if the rewards are activated previously they will not be activated again
+		rewardsActivated = rewardsActivated || networkCtx.get().areRewardsActivated();
+		rewardBalanceChanged = false;
+
 		for (int i = 0, n = pendingChanges.size(); i < n; i++) {
 			trackBalanceChangeIfAny(
 					pendingChanges.id(i).getAccountNum(),
 					pendingChanges.entity(i),
 					pendingChanges.changes(i));
 		}
 		assertZeroSum();
+		if (shouldActivateStakingRewards()) {
+			networkCtx.get().setStakingRewards(true);
+			stakingInfo.get().forEach((entityNum, info) -> info.clearRewardSumHistory());
```

Comment:
I see this in the HIP as well, but there was a meeting where I think Richard questioned if we want to do this. Do you remember that?

Reply:
I wasn't aware of that 👀